src/ik_gradient.cpp

```diff
@@ -278,16 +263,52 @@ template <size_t threads> struct IKJacobian : IKJacobianBase<IKBase>
     {
         IKJacobianBase<IKBase>::initialize(problem);
         solution = problem.initial_guess;
-        if(thread_index > 0)
+        if(thread_index_ > 0)
             for(auto& vi : problem.active_variables)
-                solution[vi] = random(modelInfo.getMin(vi), modelInfo.getMax(vi));
+                solution[vi] = random(modelInfo_.getMin(vi), modelInfo_.getMax(vi));
     }
     const std::vector<double>& getSolution() const { return solution; }
     void step() { optimizeJacobian(solution); }
     size_t concurrency() const { return threads; }
 };
-static IKFactory::Class<IKJacobian<1>> jac("jac");
-static IKFactory::Class<IKJacobian<2>> jac_2("jac_2");
-static IKFactory::Class<IKJacobian<4>> jac_4("jac_4");
-static IKFactory::Class<IKJacobian<8>> jac_8("jac_8");
+
+std::optional<std::unique_ptr<IKSolver>> makeGradientDecentSolver(
+    const IKParams& params) {
+  const auto& name = params.solver_class_name;
+  if (name == "gd")
+    return std::make_unique<IKGradientDescent<' ', 1>>(params);
+  else if (name == "gd_2")
```

Comment:
Same comment as above about not needing any of these `else` statements.

Reply:
Addressed in [e9c01ee](https://github.com/PickNikRobotics/bio_ik/pull/1/commits/e9c01ee4ef00ced5d79bb788ab01a067e93e922c)